### PR TITLE
Extract credProtect from credman response

### PR DESCRIPTION
**Pull request by FabianHenneke**
The credProtect level of a resident credential is returned under the
key 0x0A (10) of a credential management response. With this commit,
the value is parsed and can be read with fido_cred_prot.

### CODE REVIEW

File: src/credman.c [hunk 1]
```diff
@@ -249,6 +250,15 @@ credman_parse_rk(const cbor_item_t *key, const cbor_item_t *val, void *arg)
...
 		cred->type = cred->attcred.type; /* XXX */
 		return (0);
+	case 10:
+		if (cbor_decode_uint64(val, &prot) < 0)
+			return (-1);
+		if (prot != FIDO_CRED_PROT_UV_OPTIONAL &&
+		    prot != FIDO_CRED_PROT_UV_OPTIONAL_WITH_ID &&
+		    prot != FIDO_CRED_PROT_UV_REQUIRED)
+			return (-1);
+		cred->ext.prot = (int)prot;
```
Comment (martelletto) [inline src/credman.c:260]: Would it make sense to use fido_cred_set_prot() here?

Comment (FabianHenneke) [reply src/credman.c]: Agreed and force-pushed.

File: src/credman.c [hunk 2]
```diff
@@ -249,6 +250,10 @@ credman_parse_rk(const cbor_item_t *key, const cbor_item_t *val, void *arg)
 			return (-1);
 		cred->type = cred->attcred.type; /* XXX */
 		return (0);
+	case 10:
+		if (cbor_decode_uint64(val, &prot) < 0)
+			return (-1);
+		return (fido_cred_set_prot(cred, (int)prot));
```
Comment (FabianHenneke) [inline src/credman.c:256]: Is this conversion fine without range checks?

Comment (FabianHenneke) [reply src/credman.c]: I force-pushed a new version that does mimic the robust check from pin.c and turns every negative return value of fido_cred_set_prot into a -1.

**Comment by FabianHenneke** [comment]
Thanks!

**Pull request merged**